examples/bpi-monolith-test-suite/tests/zk_tests.ts

```diff
@@ -1,70 +1,131 @@
-        const agreement = new AgreementAlt("alice", "bob", ["cup1", "plate2"]);
-        const order = new OrderAlt("cup1", "cup1" + "notbob")
-        agreement.deployShield();
-        const shield = agreement.shield;
-        let [
-            agreementStateTree,
-            productTree,
-            _
-        ]: [MerkleTree, MerkleTree, MerkleTree]
-            = agreement.merkleizeState();
-        // private inputs contain all information to validate new state object and calculate new agreement root
-        let privateInput = {
-            order: order,
-            orderSalt: 'salt',
-            productIdsRoot: productTree.getRoot(),
-            productIdsProof: productTree.getProof(sha256(order.productId).toString()),
-            buyerPK: agreement.buyerPK,
-            sellerPK: agreement.sellerPK,
-            agreementStateRoot: agreementStateTree.getRoot(),
-            agreementStateSalt: 'salt',
-            salt: 'salt'
-        }
-        let orderLeaves = Object.entries(order).map(x => sha256(x[1]));
-        let orderTree = new MerkleTree(orderLeaves, sha256);
-        let orderRoot = orderTree.getRoot();
-        let orderCommitment = Buffer.from(sha256(orderRoot + 'salt').toString(), 'hex')
-        let input = {
-            agreementStateCommitment: shield.agreementStateCommitment,
-            stateObjectCommitment: orderCommitment
-        }
-        expect(() => shield.verifiers['placeOrder'].predicate(input, privateInput)).to.throw("Invalid proof")
+    it("Merkleizing the Agreement state should create a product merkle tree", () => {
+      const productRoot = productTree.getRoot().toString("hex");
+      const productLeaf = sha256("cup1");
+      const productTreeProof = productTree.getProof(productLeaf);
+
+      expect(productTree.verify(productTreeProof, productLeaf, productRoot)).to
+        .be.true;
     });
+    it("Merkleizing the Agreement state should create an empty order merkle tree", () => {
+      const orderLeaves = orderTree.getLeaves();
+      expect(orderLeaves.length).to.be.eql(0);
+    });
+    it("Merkleizing the Agreement state should create an agreement state merkle tree", () => {
+      const agreementStateRoot = agreementStateTree.getRoot().toString("hex");
+
+      //AgreementStateTree leaves include buyerPK, sellerPK, productTree root and orderTree root.
+      const agreementStateLeaf = agreement.buyerPK;
+      const agreementStateTreeProof =
+        agreementStateTree.getProof(agreementStateLeaf);
 
-});
\ No newline at end of file
+      expect(
+        agreementStateTree.verify(
+          agreementStateTreeProof,
+          agreementStateLeaf,
+          agreementStateRoot
+        )
+      ).to.be.true;
+    });
+  });
+  it("Bob places a valid order, it is added to the bpi and the committed state reflects the new BPI agreement state.", () => {
+    // Initialize
+    const agreement = new AgreementAlt("alice", "bob", ["cup1", "plate2"]);
```

Comment:
do we need to redeploy it? in before method same thing is deployed?

Reply:
Removed unnecessary code.